brainscore_vision/data/data_generation_trial/extract_nwb_data.py

```diff
@@ -0,0 +1,132 @@
+import json
+import numpy as np
+# import fsspec
+import h5py
+import remfile
+import requests
+
+from brainio.stimuli import StimulusSet
+from dandi.dandiapi import DandiAPIClient
+from IPython.display import display
+from pynwb import NWBHDF5IO
+from pynwb.base import Images
+from PIL import Image
+
+exp_name    = 'emogan' 
+nwb_name    = 'ecephys'
+subject     = 'pico'
+
+nwb_file_path   = "/Users/caroljiang/Downloads/vision/brainscore_vision/data/data_generation_trial/000788/sub-{}/sub-{}_{}.nwb".format(subject, subject, nwb_name)
+# experiment_path   = "/Users/caroljiang/Downloads/vision/brainscore_vision/data/data_generation_trial/000788/sub-{}".format(subject)
+
+### Load nwb file
+def load_nwb_file(nwb_file_path):
+    print("Loading the NWB file ...")
+    io = NWBHDF5IO(nwb_file_path, "r") 
+    nwb_file = io.read()    
+    return nwb_file
+
+def old_validate_nwb_file(nwb_file_path):
+    nwb_file = load_nwb_file(nwb_file_path)
+    if nwb_file.electrodes and 'PSTHs_QualityApproved_ZScored_SessionMerged' in nwb_file.scratch.keys():
+        return nwb_file
+    else:
+        raise ValueError('Type of NWB file not accepted, needs to contain electrode data.')
+
+def validate_nwb_file(dandiset_id, nwb_file_path):
+    nwb_file = stream_from_dandi(dandiset_id, nwb_file_path)
+    if nwb_file.electrodes and 'PSTHs_QualityApproved_ZScored_SessionMerged' in nwb_file.scratch.keys():
+        return nwb_file
+    else:
+        raise ValueError('Type of NWB file not accepted, needs to contain electrode data.')
+    
+def get_meta(nwb_file):
+    s = (nwb_file.scratch['PSTHs_QualityApproved_ZScored_SessionMerged'].description.split('[start_time_ms, stop_time_ms, tb_ms]: ')[-1])
```

Comment:
I would add docstrings and type hints, but also some error catching in the event that something (like the `split`) fails